lib/mergeDeep.js

```diff
@@ -55,6 +55,10 @@ class MergeDeep {
       return this.compareEmptyTarget(t, s, additions, modifications)
     }
 
+    if (typeof s === 'string' && (t.name === s || t.username === s)) {
```

Comment:
How about doing this instead:

```suggestion
    if (typeof s === 'string' && (Object.keys(t).some(key => NAME_FIELDS.includes(key)))) {
```

At least you get rid of the duplication of the `name` and `username` fields.

Reply:
@martinm82 This avoids looping through the characters in the given conditions:

```javascript
t = { name: "something" }
s = "something"
```

..where "something" is a team name for example, thus the reason why I compare the values.
But I think you are right, we don't need to do that here. We can use your suggestion to do the early exit no matter the value for the given object's name/username property. We only need to check the shape of the inputs. Thank you!

Reply:
I just pushed a fixup commit for this, I'll rebase and squash it into the other commit relevant to this when I have approval on the PR.

Reply:
BTW, how about adding some comment explaining the why. Today I understand this block but tomorrow I for sure won't understand the reasoning behind it. Yes, there is git blame and I probably would find this pull request again.

Reply:
Yeah, I thought the commit would explain it, but I can for sure add a comment in the code about it.

<img width="916" alt="Screenshot 2023-01-09 at 09 57 15" src="https://user-images.githubusercontent.com/435885/211271337-6aefee50-afef-4579-b453-bcfcd13c7fff.png">

Reply:
@martinm82 Another thought.. Would it ever be reasonable to continue the comparison from here if the source `s` is a string? I mean, that would never add any useful info to the arrays (additions, modifications).

What I mean is that I can simplify it down to:

```javascript
if (typeof s === 'string') {
  return
}
```

Reply:
Let me think about it but in general if we have a good test coverage and test cases are passing as expected we might indeed skip the other condition.

Reply:
@martinm82 I pushed a fixup commit now with a comment and the simplification that I mentioned above. The guard could also be placed in all locations of the recursive calls. Given the current design of the function, the recursive calls should never take place for primitives such as strings. (primitives are compared in the else block of the loop)

Another design is to recurse into the function even for primitives and move the comparison up from the else block in the loop and to this guard before the loop starts. That however would add more recursive calls but with the tradeoff that the function can be called with primitive values (such as two strings) and have the differences added to the given arrays (additions, modifications). It's however a bit weird IMHO to use the function like this with primitives. When comparing primitives it would probably make more sense with a conventional compare function that returns -1, 0, or 1.